src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -300,6 +284,56 @@ UpdateRelationsToLocalShardTables(Node *node, List *relationShardList)
 								 relationShardList, QTW_EXAMINE_RTES_BEFORE);
 	}
 
+	if (IsA(node, OnConflictExpr))
+	{
+		OnConflictExpr *onConflict = (OnConflictExpr *) node;
+		Oid constraintId = onConflict->constraint;
+
+		if (constraintId != InvalidOid)
```

Comment:
minor, in general could use OidIsValid to check for InvalidOid

Reply:
thank you. edited. it looks better now